moira/cache.py

```diff
@@ -1,9 +1,31 @@
 from functools import wraps
 
 from twisted.internet import reactor, defer
+from collections import OrderedDict
 
-CACHE = {}
+#CACHE = {}
+class CacheClass(OrderedDict):
+    """
+    Autocleaning dict.
+    """
+    
+    max_live_time = 2*60**2 # 2 hours
+    
+    def __setitem__(self, key, value, *args, **kwargs):
+        if key in self:
+            del self[key]
+        super(CacheClass, self).__setitem__(key, value, *args, **kwargs)
+    
+    def cleanup(self):
+        now = reactor.seconds()
+        seconds = now - self.max_live_time
+        
+        for key in self.iterkeys():
+            if self[key][0] > seconds:
+                break
+            del self[key]
```

Comment:
>>> o = OrderedDict(((1, 'a'), (2, 'b'), (3, 'c')))
>>> o
OrderedDict([(1, 'b'), (2, 'b'), (3, 'c')])
>>> for k in o.iterkeys(): del o[k]
>>> o
OrderedDict()
I think it's not a problem in this case, because delete not from iterator, but from dict itself. Python solve any downstream problems. Other method is use array .keys() instead of .iterkeys(), but this is real problem if dict is big.